src/main/java/bit/javaoop/NamedObject.java

```diff
@@ -0,0 +1,17 @@
+package bit.javaoop;
+
+public class NamedObject {
```

Comment:
Podoba mi się, że na własną rękę kombinowałeś z abstrakcją! Jak się nie przewrócisz to się nie nauczysz. A więc teraz trochę krytyki :)

Fajnie, że zauważyłeś coś co łączy wszystkie klasy z naszego przykładu. Zauważenie takiego czegoś jest pierwszym krokiem do zastanowienia się "hmm, a może dziedziczenie?". Jednak w naszym wypadku będzie to już zbyt duża abstrakcja. `NamedObject` opisuje wszystkie obiekty, które posiadają nazwę - jest to baaaaaardzo obszerna abstrakcja. Takich obiektów jest duuuużo. Tak dużo, że pokusiłbym się o wystawienie tezy - WSZYSTKIE obiekty mają nazwę :). Dlaczego? Bo każda klasa Javowa posiada rodzica, którym jest klasa `Object`. Ten rodzic daje nam dostęp do swoich metod, między innymi do metody `toString()`. Ta metoda w luźnej interpretacji oznacza: "Przedstaw się!". Wywołując na dowolnym obiekcie `toString()` dostajemy jego.... NAZWĘ! A więc każdy obiekt w Javie jest jakoś nazwany i nie musimy już dodatkowo samemu opakowywać ich w dodatkową klasę do tego celu.

Dodatkowo, trzeba się też zastanowić, czy takie uogólnienie ma sens.

Przykład:
Człowiek -> Zwierzę -> Organizm Wielokomórkowy -> Zbitek atomów

Człowiek ma metodę oddychaj(). Zwierzę ma metodę żyj(). Organizm wielokomórkowy produkujEnergię(). A zbitek atomów utworzWiązanie().
I teraz pytanie... czy chcemy, aby człowiek posiadał metodę utworzWiązanie?? :)

Przykład łopatologiczny, ale myślę, że wyjaśnia, że w pewnym momencie abstrakcja może wymknąć się spod kontroli i trzeba spojrzeć logicznie, w jakimś punkcie się zatrzymać i pomyśleć czy to ma sens. 

Reasumując: bardzo dobrze, że próbujesz i wpadłeś na taki pomysł, bo teraz mogłem Cię skrytykować i może coś z tego wyciągniesz :)